### DISCUSSION TITLE
Discrete variables

### DISCUSSION THREAD

**Discussion by Matty19941373** [anyoptimization/pymoo#379] [Q&A]
Hi all,
I have a multi objective optimization problem with the below discrete variables:

X1={"EE":10, "DWG":20,"C":50}
X2={"EE":2, "DWG":0.2,"C":54}
X3={"EE":15, "DWG":4,"C":15}
X4=[40,50,60,70,80]
X5=[20,30]

Note that the elements of X1, X2, and X3 cannot be combined with each other that's why i defined them in a dictionary to describe them. But X1, X2, and X3 can be combined with different elements of X4, and X5. To elaborate, one feasible combination can be as follows:
X_sample=[{"EE":10, "DWG":20,"C":50},40,30]

but the [{"EE":10, "DWG":4,"C":54},40,30] is infeasible as it combines elements of the dictionary together (EE's value from X1, DWG's value from X3, and C's value from X2). 

there are three objective function which take these variables as inputs.

I could not implement this problem in pymoo. 
Can you please assist?

Thanks

**Comment by blankjul** [comment]
Hey! Does this help you out: https://pymoo.org/customization/mixed.html

It looks like to me you have 5 choice variables. For X1-3 you can have the mapping from string to integer directly in the `_evaluate` function. I am not sure if this Is just a toy example, but otherwise, the search space is quite small, so you can even do an exhaustive search and get the true optimum.
